### PR TITLE
Added A4 page size. Set max page size as default option for flatbed a…

### PR DESCRIPTION
**Pull request by rquast**
…nd document feeder. Set default resolution to 300 DPI.

### CODE REVIEW

File: scanline/AppController.m [hunk 1]
```diff
@@ -538,15 +538,17 @@ - (IBAction)startScan:(id)sender
     
     if ( ( fu.scanInProgress == NO ) && ( fu.overviewScanInProgress == NO ) )
     {
-        if ( fu.type == ICScannerFunctionalUnitTypeDocumentFeeder )
-        {
-            ICScannerFunctionalUnitDocumentFeeder* dfu = (ICScannerFunctionalUnitDocumentFeeder*)fu;
+
+        ICScannerFunctionalUnitDocumentFeeder* dfu = (ICScannerFunctionalUnitDocumentFeeder*)fu;
```
Comment (klep) [inline scanline/AppController.m]: Doesn't this explicit cast assume the functional unit is a document feeder?

File: scanline/ScanConfiguration.m [hunk 2]
```diff
@@ -76,7 +79,7 @@ + (NSDictionary*)configOptions
                      @"synonyms": @[@"res", @"minResolution"],
                      @"type": @"string",
                      @"description": @"Specify minimum resolution at which to scan (in dpi)",
-                     @"default": @"150"
+                     @"default": @"300"
```
Comment (klep) [inline scanline/ScanConfiguration.m]: Why 300? Seems as arbitrary as 150 :)

**Comment by rquast** [comment]
Closing this PR. Sorry about all of this.. I was a bit excited and didn't test anything either. You're right about both things.

**Comment by rquast** [comment]
For most of the issues I've opened, I had a think more about them. I think I'd prefer to pipe in json actions and configuration, and as a result of doing that, it sends me back json. This is a much more simple and flexible option for me to do some IPC with the program rather than using command lines without structured output. It also means you don't have to change any of the existing commands or settings. Does this sound okay?

**Comment by rquast** [comment]
Can also use json streaming for stdout progress updates. https://en.wikipedia.org/wiki/JSON_Streaming

File: scanline/AppController.m (hunk 1 above)
Comment (rquast) [inline scanline/AppController.m]: Sorry, this should check for the flatbed type and cast.. ICScannerFunctionalUnitTypeFlatbed

**Comment by rquast** [comment]
Have started a json version of this from scratch to see how I go.. it will use the deserialized json file as the config, instead of scannerconfiguration, so I figured it's easier to rewrite. https://github.com/rquast/jsonscan

**Pull request closed without merging**